apis/v1alpha1/types.go

```diff
@@ -0,0 +1,661 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package v1alpha1
+
+import (
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	"github.com/aws/aws-sdk-go/aws"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Hack to avoid import errors during build...
+var (
+	_ = &metav1.Time{}
+	_ = &aws.JSONValue{}
+	_ = ackv1alpha1.AWSAccountID("")
+)
+
+// This structure specifies the VPC subnets and security groups for the task,
+// and whether a public IP address is to be used. This structure is relevant
+// only for ECS tasks that use the awsvpc network mode.
+type AWSVPCConfiguration struct {
+	AssignPublicIP *string   `json:"assignPublicIP,omitempty"`
+	SecurityGroups []*string `json:"securityGroups,omitempty"`
+	Subnets        []*string `json:"subnets,omitempty"`
+}
+
+// The array properties for the submitted job, such as the size of the array.
+// The array size can be between 2 and 10,000. If you specify array properties
+// for a job, it becomes an array job. This parameter is used only if the target
+// is an Batch job.
+type BatchArrayProperties struct {
+	Size *int64 `json:"size,omitempty"`
+}
+
+// The overrides that are sent to a container.
+type BatchContainerOverrides struct {
+	Command              []*string                   `json:"command,omitempty"`
+	Environment          []*BatchEnvironmentVariable `json:"environment,omitempty"`
+	InstanceType         *string                     `json:"instanceType,omitempty"`
+	ResourceRequirements []*BatchResourceRequirement `json:"resourceRequirements,omitempty"`
+}
+
+// The environment variables to send to the container. You can add new environment
+// variables, which are added to the container at launch, or you can override
+// the existing environment variables from the Docker image or the task definition.
+//
+// Environment variables cannot start with "Batch". This naming convention is
+// reserved for variables that Batch sets.
+type BatchEnvironmentVariable struct {
+	Name  *string `json:"name,omitempty"`
+	Value *string `json:"value,omitempty"`
+}
+
+// An object that represents an Batch job dependency.
+type BatchJobDependency struct {
+	JobID *string `json:"jobID,omitempty"`
+	Type  *string `json:"type_,omitempty"`
```

Comment:
@RedbackThomson I am not sure I like those `type_` JSON struct fields from a UX perspective (99% users will get this wrong). Guess this is because of the JSON `type` keyword. How do we handle such cases? cc/ @A-Hilaly

Reply:
Correct, this is a keywording issue. I also don't like it, but I don't know any alternatives that could work?

Reply:
Hm, ok good to know. One option for those cases would be to prefix tags it with `Object` e.g., `ObjectType` (or `Resource`).

Reply:
Keeping it for now until we see user issues.